File: smart-poultry-dashboard/src/types/farm.ts
export interface Farm {
  id: string;
  name: string;
  location?: string | null;
  created_at: string;
}

export interface FarmCreate {
  name: string;
  location?: string | null;
}

export interface FarmUpdate {
  name?: string;
  location?: string | null;
}

File: app/schemas/farm.py
from datetime import datetime
from pydantic import BaseModel, Field


class FarmBase(BaseModel):
    name: str = Field(..., min_length=1)
    location: str | None = None


class FarmCreate(FarmBase):
    pass


class FarmUpdate(BaseModel):
    name: str | None = None
    location: str | None = None


class FarmOut(FarmBase):
    id: str
    created_at: datetime

    class Config:
        from_attributes = True
